File: modules/AcroStack.Chat/Dtos/SearchMessagesInput.cs
namespace AcroStack.Chat;

/// <summary>Input for searching chat messages by keyword across all conversations.</summary>
public class SearchMessagesInput
{
    /// <summary>Keyword to search for in message text.</summary>
    public string Keyword { get; set; } = string.Empty;

    public int SkipCount { get; set; }

    public int MaxResultCount { get; set; } = 50;
}

File: react/src/api/models/acroStack/chat/SearchMessagesInput.ts
/* oxlint-disable */

/**
 * @description Input for searching chat messages by keyword across all conversations.
 * @type object
 */
export type AcroStackChatSearchMessagesInput = {
  /**
   * @description Keyword to search for in message text.
   * @type string | undefined
   */
  keyword?: string | null;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  skipCount?: number;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  maxResultCount?: number;
};
